packages/cli/src/commands/generate/function/function.js

```diff
@@ -3,48 +3,58 @@ import path from 'path'
 import camelcase from 'camelcase'
 import terminalLink from 'terminal-link'
 
-import { getPaths } from 'src/lib'
+import { getPaths, transformTSToJS } from 'src/lib'
 
+import { yargsDefaults } from '../../generate'
 import {
   templateForComponentFile,
   createYargsForComponentGeneration,
 } from '../helpers'
 
-export const files = async ({ name, ...rest }) => {
+export const files = ({ name, ...rest }) => {
+  // Taken from ../component; should be updated to take from the project's configuration
+  const isJavascript = rest.javascript && !rest.typescript
+
   const functionName = camelcase(name)
   const file = templateForComponentFile({
     name: functionName,
     componentName: functionName,
+    extension: isJavascript ? '.js' : '.ts',
     apiPathSection: 'functions',
     generator: 'function',
-    templatePath: 'function.js.template',
+    templatePath: 'function.ts.template',
     templateVars: { ...rest },
-    outputPath: path.join(getPaths().api.functions, `${functionName}.js`),
+    outputPath: path.join(getPaths().api.functions, `${functionName}${isJavascript ? '.js' : '.ts'}`),
   })
 
-  return { [file[0]]: file[1] }
+  // If --javascript=true convert TS->JS before returning
```

Comment:
I think this comment is redundant. It's clear from the code what's going on here